File: contractstudy/src/test/java/contractstudy/hierarchy/projectClassExtractor/java/testData/SubClass1.java
package contractstudy.hierarchy.projectClassExtractor.java.testData;

import java.util.Arrays;
import java.util.Objects;

enum SubEnum1 {
  NICE, COOL;
  String text() {return "True";};
}

abstract class SubAbstractClass1 {
  public String test() {
    return "test";
  }
}

public class SubClass1 extends ParentClass implements ParentInterface {

  public SubClass1() {
    super("test");
  }

  @Override
  public String test() {
    super.test();
    return "test";
  }

  @Override
  public String test2(String param1) {
    return param1;
  }

  public int test3(int param1, int param2) {
    int sum = super.testSum(param1, param2);
    return param1 + sum;
  }

  public String test3(String[] text) {
    return Arrays.stream(text).filter(x -> Objects.equals(x, "Test")).toString();
  }

  @Override
  public String printName() {
    return null;
  }
}

File: contractstudy/src/test/java/contractstudy/hierarchy/projectClassExtractor/kotlin/testData/SubClass1.kt
package contractstudy.hierarchy.projectClassExtractor.kotlin.testData

import java.util.*

enum class Cool {
    Nice, Cool, Super
}

class SubClass1 : ParentClass, ParentInterface {

    constructor(text: String) : super(text)

    override fun test(): String {
        super.test()
        return "test"
    }

    override fun test2(param1: String): String {
        return param1
    }

    fun test3(param1: Int, param2: Int): Int {
        val sum = super.testSum(param1, param2)
        return param1 + sum
    }

    fun test3(text: Array<String>?): String {
        return Arrays.stream(text).filter { x: String? ->
            x == "Test"
        }.toString()
    }

    override fun printName(): String {
        return "Text"
    }
}